www/src/components/site-metadata.js

```diff
@@ -31,7 +31,7 @@ const SiteMetadata = ({ pathname, locale }) => {
         content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"
       />
 
-      <meta property="og:url" content={siteUrl} />
+      <meta property="og:url" content={`${siteUrl}${pathname}`} />
```

Comment:
Instead of `${siteUrl}${location.pathname}`, you can use `location.href` here. Change the props to take in `href` instead of `pathname` and edit `<Layout>` to match.

Reply:
@tesseralis good idea. However with a url such as 
www.gatsbyjs.com/showcase/www.shopflamingo.com location.href is undefined. 

So should the meta og:url be like this?

```javascript
 <meta property="og:url" content={`${ typeof location.href !== "undefined" ? location.href : `${siteUrl}${location.pathname}` }`} />
```

which would return this if undefined

```html
https://www.gatsbyjs.org/showcase

```

Reply:
`location.href` is not available for SSR -> `location.pathname` is available. 
(https://github.com/gatsbyjs/gatsby/pull/21815#discussion_r387356011)